### PR TITLE
Compute top people locally

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Remove `TopConversationsRequestStrategy` and compute the top people locally.
* The computation is based on the number of messages that are in a `.oneOnOne` conversation.
* The number of top people is limited to 25 at the moment.
* Fix a test related to notifications which was broken in https://github.com/wireapp/wire-ios-sync-engine/pull/277.

### CODE REVIEW

File: Tests/Source/UserSession/ZMStoredLocalNotificationTests.m [hunk 1]
```diff
@@ -73,7 +73,7 @@ - (void)testThatItCreatesAStoredLocalNotificationFromALocalNotification
     // then
     XCTAssertEqual(storedNote.conversation, self.conversation);
     XCTAssertEqualObjects(storedNote.senderUUID, self.sender.remoteIdentifier);
-    XCTAssertEqualObjects(storedNote.category, ZMConversationCategory);
+    XCTAssertEqualObjects(storedNote.category, ZMConversationCategoryIncludingLike);
```
Comment (daehn) [inline Tests/Source/UserSession/ZMStoredLocalNotificationTests.m]: This test was broken in https://github.com/wireapp/wire-ios-sync-engine/pull/277

File: Source/UserSession/Search/TopConversationsDirectory.swift [hunk 2]
```diff
@@ -44,17 +45,36 @@ private let topConversationsObjectIDKey = "WireTopConversationsObjectIDKey"
...
+    private func updateUIList(with identifiers: [NSManagedObjectID]) {
+        uiMOC.performGroupedBlock {
+            self.topConversationsCache = identifiers.flatMap {
+                (try? self.uiMOC.existingObject(with: $0)) as? ZMConversation
+            }
             self.persistList()
         }
     }
+
+    private func messageCount(for conversation: ZMConversation) -> Int {
```
Comment (mikeger) [inline Source/UserSession/Search/TopConversationsDirectory.swift]: Could be a method in extension on ZMConversation?

Comment (daehn) [inline Source/UserSession/Search/TopConversationsDirectory.swift]: @mikeger done

**Comment by marcoconti83** [comment]
Any indication on the performance?

**Comment by mikeger** [comment]
@marcoconti83 @daehn please merge when you are ready

**Comment by daehn** [comment]
@marcoconti83 when testing on device there it didn't look like there are any performance issues.

**Comment by teller** [comment]
#bossmode

**Pull request merged**